lib/LaTeXML/Core/Rewrite.pm

```diff
@@ -479,7 +479,14 @@ sub domToXPath_rec {
   }
 
   elsif ($type == XML_TEXT_NODE) {
-    return ("*[text()='" . $node->textContent . "']", 1, 0); } }
+    return ("*[text()=" . quoteXPathLiteral($node->textContent) . "]", 1, 0); } }
+
+# Return quoted string, but note: XPath doesn't provide sensible way to slashify ' or "
+sub quoteXPathLiteral {
+  my ($string) = @_;
+  if    ($string !~ /'/) { return "'" . $string . "'"; }
+  elsif ($string !~ /"/) { return '"' . $string . '"'; }
+  else { return 'concat(' . join(',"\'",', map { "'" . $_ . "'"; } split(/'/, $string)) . ')'; } }
```

Comment:
I think you could get by without the third case, though this code looks quite workable. If you feel like a slightly simpler idea, you could standardize to always use `"` as the xpath delimiter and have a single replacement call of:

` $string =~ s/([^\\])"/$1\\"/g;`

whenever an xpath string is being evaluated inside the value of a key="value".

Reply:
quotemeta is interesting, but we're so often doing quoting from one language to another, which screws things up; you have to know the quoting conventions involved.  Case in point: XPath 1 doesn't provide for slashifying single and double quotes. If you have a string with both, you've got to use clumsy code like the above.

Reply:
I think (hope) we're there.  I'm going to go ahead & merge, since there's some DLMF stuff waiting on it.  Thanks for your observations!

Reply:
Great!